File: test_api.dart
// ignore_for_file: avoid_print
import 'package:dio/dio.dart';

void main() async {
  final dio = Dio(BaseOptions(
    baseUrl: 'http://127.0.0.1:8000/api/v1',
    responseType: ResponseType.json,
  ),);

  // Enable logging
  dio.interceptors.add(LogInterceptor(
    requestBody: true,
    responseBody: true,
    error: true,
  ),);

  try {
    print('--- TESTING CHECKOUT API ---');
    final payload = {
      'items': [
        {'product_id': '1234', 'quantity': 1},
      ],
      'delivery_address': 'Tashkent',
      'payment_method': 'Cash',
      'delivery_method': 'Delivery Service',
      'customer_notes': 'Call me',
    };
    final response = await dio.post('/checkout/order', data: payload);
    print('Checkout response: ${response.statusCode} ${response.data}');
  } on DioException catch (e, _) {
    print('Checkout DioError: ${e.response?.statusCode}');
    print('Checkout Error details: ${e.response?.data}');
  }

  try {
    print('\n--- TESTING PRODUCTS API WITH CAMEL CASE PARAMS ---');
    final response = await dio.get('/products', queryParameters: {
      'minPrice': 5000,
      'maxPrice': 10000,
    },);
    print('Products response: ${response.statusCode} ${response.data}');
  } on DioException catch (e, _) {
    print('Products DioError: ${e.response?.statusCode}');
    print('Products Error details: ${e.response?.data}');
  }
}

File: backend/test_api.py
import sys
import asyncio
from fastapi.testclient import TestClient
from app.main import app
from app.models.user import User
from app.api.deps import get_current_user
import uuid

async def override_get_current_user():
    return User(id=uuid.uuid4(), phone="+998901234567")

app.dependency_overrides[get_current_user] = override_get_current_user

client = TestClient(app)

response = client.post('/api/v1/checkout/order', json={'items': [{'product_id': '123e4567-e89b-12d3-a456-426614174000', 'quantity': 1}], 'delivery_address': 'Tashkent', 'payment_method': 'Click', 'delivery_method': 'Delivery', 'customer_notes': 'Call me'})
print("CHECKOUT RESPONSE:", response.status_code, response.json())

response = client.post('/api/v1/checkout/order', json={'items': [{'product_id': 'not-a-uuid', 'quantity': 1}], 'delivery_address': 'Tashkent', 'payment_method': 'Click', 'delivery_method': 'Delivery', 'customer_notes': 'Call me'})
print("CHECKOUT INVALID UUID RESPONSE:", response.status_code, response.json())

response = client.get('/api/v1/products?minPrice=5000&maxPrice=10000')
print("PRODUCTS PARAMS ERROR RESPONSE:", response.status_code, response.json())

response = client.get('/api/v1/products?min_price=5000&max_price=10000')
print("PRODUCTS FIXED PARAMS RESPONSE:", response.status_code, len(response.json().get('data', [])))

response = client.get('/api/v1/products')
print("PRODUCTS NO PARAMS ERROR RESPONSE:", response.status_code, len(response.json().get('data', [])))
